integration-tests/keycloak-authorization/src/test/java/io/quarkus/it/keycloak/KeycloakLifecycleManager.java

```diff
@@ -45,22 +44,16 @@ public class KeycloakLifecycleManager implements QuarkusTestResourceLifecycleMan
     protected static String KEYCLOAK_SERVER_URL;
     private static final String KEYCLOAK_REALM = "quarkus";
     private static final String KEYCLOAK_SERVICE_CLIENT = "quarkus-service-app";
-    private static final String KEYCLOAK_IMAGE = System.getProperty("keycloak.docker.image");
+    private static final String KEYCLOAK_VERSION = System.getProperty("keycloak.version");
 
     @SuppressWarnings("resource")
     @Override
     public Map<String, String> start() {
-        try {
-            keycloak = new GenericContainer<>(
-                    new ImageFromDockerfile().withDockerfile(Paths.get(getClass().getResource("/Dockerfile").toURI()))
-                            .withBuildArg("KEYCLOAK_IMAGE", KEYCLOAK_IMAGE))
-                    .withExposedPorts(8080)
-                    .withEnv("KEYCLOAK_ADMIN", "admin")
-                    .withEnv("KEYCLOAK_ADMIN_PASSWORD", "admin")
-                    .waitingFor(Wait.forLogMessage(".*Keycloak.*started.*", 1));
-        } catch (URISyntaxException e) {
-            throw new RuntimeException(e);
-        }
+        keycloak = new GenericContainer<>("quay.io/keycloak/keycloak:" + KEYCLOAK_VERSION)
+                .withExposedPorts(8080)
+                .withEnv("KEYCLOAK_ADMIN", "admin")
+                .withEnv("KEYCLOAK_ADMIN_PASSWORD", "admin")
+                .waitingFor(Wait.forLogMessage(".*Keycloak.*started.*", 1));
```

Comment:
Nice cleanup but I have a stupid question: can't we use the Dev Service here?

Reply:
It is a good question, we have some integration tests (everything in `integration-tests/oidc-code-flow`, `integration-tests/oidc-tenancy`, and one or two of the other smaller OIDC integration test modules - including those in quickstarts) which use the Dev Service.

However, some tests continue testing against the legacy KC WildFly based distribution, to keep some coverage for it as WildFly distribution is still likely in use, but especially `keycloak-authorization` tests as RH-SSO is based on earlier KC version and we've also had a few cases where upstream `keycloak-authorization`changes caused problems at the RHBQ level.

Does it sound reasonable ?

Reply:
That can change later thought once KC 22.x based products will go mainstream

Reply:
Yeah, it would be nice to have this "two-way" test coverage. This is especially true for Quarkus/RHBQ -> Wildfly because, product-wise, people are still going to migrate. We are doing the same on our side.

In regards to authorization, I would say we can just use dev services. If it is not a problem for RHBQ.

Reply:
@pedroigor Thanks,

>In regards to authorization, I would say we can just use dev services. If it is not a problem for RHBQ.

Do you mean that we change this test to use Dev Service ? I'm not sure it will be possible, at least for now as `Dev Service` can't do
```
keycloak
                .withCopyToContainer(Transferable.of(createPoliciesJar().toByteArray()), "/opt/keycloak/providers/policies.jar")
```.

In general, I'd keep this test running against the legacy version, it will be in line with #35474 :-). For now at least as it is with `keycloak-authorization` that we've had problems before when running upstream tests against Keycloak 18 for example when doing the release builds

Reply:
Hey @pedroigor 

>keycloak
                .withCopyToContainer(Transferable.of(createPoliciesJar().toByteArray()), "/opt/keycloak/providers/policies.jar")

This in itself is probably not a major technical issue, we can fix it for sure, but I think

#21276

would have to go first :-), i.e, we'd need to offer keycloak-authorization specific devservices support and support loading policies.

I can also create a copy of `keycloak-authorization` tests to run against the latest release of Keycloak - but that would likely have to be done as part of #35474 and without the Dev Service for now until and if we decide to do some dev service support for `keycloak-authorization` (referring to your comment at #35474).

Reply:
True. Let's keep it as is then. My point is we can do whatever is simpler and works for Quarkus.